dispatches/unit_models/concrete_tes.py

```diff
@@ -0,0 +1,964 @@
+                    property_package=property_package,
+                    has_pressure_change=self.config.has_pressure_change,
+                    segments_set=m.parent_block().segments,
+                    operating_mode="charge",
+                )
+
+                @m.Constraint(m.parent_block().segments)
+                def temperature_equality_constraints_charge(blk, s):
+                    return blk.tube_charge.hex[s].temperature_wall == blk.concrete.temperature[s]
+
+            # Add the tube side discharge model
+            if operating_mode == "discharge" or operating_mode == "combined":
+                m.tube_discharge = TubeSideHex(
+                    property_package=property_package,
+                    has_pressure_change=self.config.has_pressure_change,
+                    segments_set=m.parent_block().segments,
+                    operating_mode="discharge",
+                )
+
+                @m.Constraint(m.parent_block().segments)
+                def temperature_equality_constraints_discharge(blk, s):
+                    return blk.tube_discharge.hex[s].temperature_wall == blk.concrete.temperature[s]
+
+            # Linking constraint for the net Q_fluid var in the concrete model
+            # Q_fluid = Q_charge - Q_discharge
+            @m.Constraint(m.parent_block().segments)
+            def heat_balance_constraints(blk, s):
+                if operating_mode == "charge":
+                    return blk.concrete.heat_rate[s] == -blk.tube_charge.hex[s].heat_duty[0]
+                elif operating_mode == "discharge":
+                    return blk.concrete.heat_rate[s] == -blk.tube_discharge.hex[s].heat_duty[0]
+                elif operating_mode == "combined":
+                    return (blk.concrete.heat_rate[s] == -blk.tube_charge.hex[s].heat_duty[0]
+                            - blk.tube_discharge.hex[s].heat_duty[0])
+
+            return m
+
+        _add_inlet_outlet_ports(self)
+
+        # Add constraints connecting different time periods
+        @self.Constraint(self.time_periods, self.segments)
+        def initial_temperature_constraints(blk, p, s):
+            if p == 1:
+                return Constraint.Skip
+            return blk.period[p].concrete.init_temperature[s] == blk.period[p - 1].concrete.temperature[s]
+
+        # Add the heat transfer coefficient surrogate model
+        def htc_surrogate():
+            # TODO: Need to declare a variable for outer radius of the
+            #       concrete block and do the following calculation in a constraint
+            # Also, do this inside each hex model instead of doing it
+            # outside. Reduces a lot of additional constraints!
+            face_area = data["face_area"]
+            a = data["tube_diameter"] / 2  # Inner radius of the concrete block
+            b = sqrt(face_area / Constants.pi + a ** 2)  # Outer radius
+            k_red = 0.8  # Reduction factor for thermal conductivity
```

Comment:
Add a reference for the value, if there is one, else if it is just an assumption then you can state that.